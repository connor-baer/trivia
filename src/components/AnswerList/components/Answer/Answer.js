import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import { RadioButton } from '@sumup/circuit-ui';

const baseStyles = ({ theme }) => css`
  width: 50%;
  padding: 0 ${theme.spacings.kilo} ${theme.spacings.kilo} 0;
  cursor: pointer;

  &::before,
  &::after {
    content: none;
  }
`;

const Answer = styled(RadioButton)(baseStyles);

Answer.propTypes = {
  example: PropTypes.string
};

Answer.defaultProps = {};

/**
 * @component
 */
export default Answer;
