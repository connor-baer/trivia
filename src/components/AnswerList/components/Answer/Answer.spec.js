import React from 'react';

import Answer from './Answer';

const defaultProps = {
  value: '1',
  name: 'group',
  children: 'Foo',
  onChange: jest.fn()
};

describe('Answer', () => {
  it('should render with default styles', () => {
    const { container } = render(<Answer {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it.skip('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Answer {...defaultProps} />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
