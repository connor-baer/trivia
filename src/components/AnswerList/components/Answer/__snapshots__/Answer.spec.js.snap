// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should render with default styles 1`] = `
.emotion-0 {
  border: 0px;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0px;
  position: absolute;
  white-space: nowrap;
  width: 1px;
  width: 50%;
  padding: 0 12px 12px 0;
  cursor: pointer;
}

.emotion-0:focus + label::before {
  border-width: 2px;
  border-color: #3388FF;
}

.emotion-0::before,
.emotion-0::after {
  content: none;
}

<input
  class="emotion-0 emotion-1"
  id="radio-button_1"
  name="group"
  type="radio"
  value="1"
/>
`;
